onnxruntime/python/tools/quantization/qdq_quantizer.py

```diff
@@ -181,137 +203,182 @@ def try_replacing_upstream_output(self, upstream_output_name, output_name):
+                self.model.replace_node_input(node, tensor_name, tensor_name_dequant_output_postfix)
+                if i == 0:
                     quantized_value = QuantizedValue(
                         tensor_name,
-                        dq_output,
+                        tensor_name_dequant_output_postfix,
                         scale_name,
                         zp_name,
                         QuantizedValueType.Input,
                     )
                     self.quantized_value_map[tensor_name] = quantized_value
+        else:
+            q_input = tensor_name
+            dq_output = add_dequant_output_suffix(tensor_name)
+            if self.model.is_graph_output(tensor_name):
+                q_input = add_quant_input_suffix(tensor_name)
+                dq_output = tensor_name
+                self.model.replace_output_of_all_nodes(tensor_name, q_input)
+            else:
+                self.model.replace_input_of_all_nodes(tensor_name, dq_output)
+
+            self._create_qdq_nodes(
+                q_input,
+                add_quant_output_suffix(tensor_name),
+                add_quant_suffix(tensor_name),
+                add_quant_output_suffix(tensor_name),
+                dq_output,
+                add_dequant_suffix(tensor_name),
+                scale_name,
+                zp_name,
+            )
+
+            quantized_value = QuantizedValue(
+                tensor_name,
+                dq_output,
+                scale_name,
+                zp_name,
+                QuantizedValueType.Input,
+            )
+            self.quantized_value_map[tensor_name] = quantized_value
+
+    def _quantize_normal_tensors(self):
+        for tensor_name, tensor_info in self.tensors_to_quantize.copy().items():
+            if tensor_name in self.quantized_value_map.keys():
+                continue
+
+            if tensor_info.tensor_type == QDQQuantTensorType.NORMAL:
+                # Quantize the input
+                initializer = find_by_name(tensor_name, self.model.initializer())
+                if initializer:
+                    self._add_qdq_pair_for_weight(initializer, tensor_info.axis)
                 else:
-                    q_input = tensor_name
-                    q_output = tensor_name + "_QuantizeLinear"
-                    dq_input = q_output
-                    dq_output = tensor_name + "_DequantizeLinear"
-                    if self.model.is_graph_output(tensor_name):
-                        q_input = tensor_name + "_QuantizeLinearInput"
-                        dq_output = tensor_name
-                        self.model.replace_output_of_all_nodes(tensor_name, q_input)
+                    data_found, scale_name, zp_name, _, _ = self._get_quantization_params(tensor_name)
+
+                    if data_found == False:
+                        raise ValueError(
+                            "Quantization parameters are not specified for param {}. "
```

Comment:
C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
Raw output